libminifi/src/utils/SystemCPUUsageTracker.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utils/SystemCPUUsageTracker.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+#ifdef __linux__
+
+SystemCPUUsageTracker::SystemCPUUsageTracker() :
```

Comment:
as the functionality here is inherently platform specific we could create separate `SystemCPUUsageTracker{Linux,Win,Apple}.cpp` files and include them here (they would need to be placed in a `platform` directory as the `cpp` files are collected by a glob)

alternatively we could also make the distinction at the cmake level

Reply:
Yeah that would probably make it easier to maintain.
There are however no current patterns on these and I am not sure which version would be better, currently there are also couple of functions which are similarly handled (#ifdef separated by platform in the function body) (e.g. FileUtils.h get_executable_path)

Maybe we could file a followup Jira where we can discuss how to handle these platformspecific codes? (CPUUsageTrackers, OsUtils, FileUtils etc)